apollo-router/src/plugins/telemetry/dynamic_attribute.rs

```diff
@@ -223,22 +223,26 @@ impl EventDynAttribute for ::tracing::Span {
                         if key.as_str().starts_with(APOLLO_PRIVATE_PREFIX) {
                             return;
                         }
-                        let mut extensions = s.extensions_mut();
-                        match extensions.get_mut::<OtelData>() {
-                            Some(otel_data) => match &mut otel_data.event_attributes {
-                                Some(attributes) => {
-                                    attributes.insert(key, value);
-                                }
+                        if s.is_sampled() {
+                            let mut extensions = s.extensions_mut();
+                            match extensions.get_mut::<OtelData>() {
+                                Some(otel_data) => match &mut otel_data.event_attributes {
+                                    Some(attributes) => {
+                                        attributes.insert(key, value);
+                                    }
+                                    None => {
+                                        let mut order_map = OrderMap::new();
+                                        order_map.insert(key, value);
+                                        otel_data.event_attributes = Some(order_map);
+                                    }
+                                },
                                 None => {
-                                    let mut order_map = OrderMap::new();
-                                    order_map.insert(key, value);
-                                    otel_data.event_attributes = Some(order_map);
+                                    // Can't use ::tracing::error! because it could create deadlock on extensions
+                                    eprintln!("no OtelData, this is a bug");
                                 }
-                            },
-                            None => {
-                                // Can't use ::tracing::error! because it could create deadlock on extensions
-                                eprintln!("no OtelData, this is a bug");
                             }
+                        } else {
+                            // FIXME: we should put event attributes somewhere else to make it work even if it's not sampled like we did with LogAttributes
```

Comment:
I created the [related issue ](https://github.com/apollographql/router/issues/5448)